nhr-loader/src/main/java/nl/b3p/brmo/nhr/loader/cli/NHRLoadUtils.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.cxf.ws.addressing.JAXWSAConstants;
+import org.apache.cxf.ws.security.wss4j.WSS4JOutInterceptor;
+import org.apache.wss4j.common.ext.WSPasswordCallback;
+import org.apache.wss4j.dom.handler.WSHandlerConstants;
+
+
+public class NHRLoadUtils {
+    public static BrmoFramework getFramework(NHRDatabaseOptions databaseOptions) throws BrmoException {
+        BasicDataSource dsStaging = new BasicDataSource();
+
+        dsStaging.setUrl(databaseOptions.getConnectionString());
+        dsStaging.setUsername(databaseOptions.getUser());
+        dsStaging.setPassword(databaseOptions.getPassword());
+
+        BrmoFramework fw = new BrmoFramework(dsStaging, null);
+        fw.setOrderBerichten(true);
+        return fw;
+    }
+
+    private static Properties getCryptoProperties(NHRCertificateOptions certificateOptions) {
+        Properties props = new Properties();
+
+        props.setProperty("org.apache.ws.security.crypto.provider", "org.apache.ws.security.components.crypto.Merlin");
+        props.setProperty("org.apache.ws.security.crypto.merlin.keystore.file", certificateOptions.getKeystore());
+        props.setProperty("org.apache.ws.security.crypto.merlin.keystore.password", certificateOptions.getKeystorePassword());
+        props.setProperty("org.apache.ws.security.crypto.merlin.keystore.type", "PKCS12");
+        props.setProperty("org.apache.ws.security.crypto.merlin.keystore.alias", certificateOptions.getKeystoreAlias());
+
+        return props;
+    }
+
+    public static Dataservice getDataservice(String targetLocation, boolean preprod, NHRCertificateOptions certificateOptions) throws Exception {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+
+        DataserviceService dataServiceService = new DataserviceService();
+        Dataservice dataService = dataServiceService.getDataserviceSoap11();
+        Client client = ClientProxy.getClient(dataService);
+        Endpoint endpoint = client.getEndpoint();
+        HTTPConduit http = (HTTPConduit) client.getConduit();
+
+        // Set up TLS certificates/keys
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("SunX509");
+        KeyStore trustStore = KeyStore.getInstance("PKCS12");
+        trustStore.load(new FileInputStream(certificateOptions.getTruststore()), certificateOptions.getTruststorePassword().toCharArray());
+        trustManagerFactory.init(trustStore);
+
+        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+        KeyStore keyStore = KeyStore.getInstance("PKCS12");
+        keyStore.load(new FileInputStream(certificateOptions.getKeystore()), certificateOptions.getKeystorePassword().toCharArray());
+        keyManagerFactory.init(keyStore, certificateOptions.getKeystorePassword().toCharArray());
+
+        SSLContext context = SSLContext.getInstance("TLSv1.2");
```

Comment:
moet dit specifiek versie 1.2 zijn? Oja het zijn windows servers natuurlijk..

Reply:
Dit zou ik kunnen veranderen in `"TLS"`, maar de aansluitdocumentatie zegt specifiek "TLS 1.2", dus hier aan houden maakt niet te veel uit.